x-pack/plugins/cloud/server/plugin.ts

```diff
@@ -40,6 +40,14 @@ export interface CloudSetup {
    * The full URL to the Kibana deployment.
    */
   kibanaUrl?: string;
+  /**
+   * The full URL to the serverless projects.
+   */
+  projectsUrl?: string;
```

Comment:
Looking at this naming I'm not convinced it is 100% correct as we default it to a path, not a URL in our serverless config 🤔 , will it actually be a full URL in our serverless configuration? CC @Dosant

Reply:
I'm wondering if these should go under serverless property, provided that they are serverless specific.

Reply:
Yeah I was also thinking that... It's currently not scoped that way in the cloud plugins' `config.ts` as there are a number of URL values not scoped to `serverless` key either... I think it may already be in use too so I figured we'd mirror that structure in our contract.

Don't feel strongly about it, so I'll go with your opinion if you disagree with my reasoning?

Reply:
Looked at each of the 2:
* `baseUrl`: I'm fine with this one being top level. It exists for more than 2 years and does not seem to be serverless specific. Perhaps we could update the description to match that [of the config object](https://github.com/elastic/kibana/blob/main/x-pack/plugins/cloud/README.md#baseurl).
* `projectsUrl`: Added yesterday through [this PR](https://github.com/elastic/kibana/pull/163076/files#diff-ed856825e1d872d57b84908c2b0523bee3b567df3cb5f98188119010f4493d88). This one seems to be serverless specific, and since 8.10.0 hasn't been released yet, perhaps we can challenge the fact that it is not under the serverless. CC @Dosant WDYT?

Reply:
> BTW the property is not present in the README.md.

sidenote on the readme: seems like we can capture this info in one place: the TS interface itself 😅

Reply:
@gsoldevila, 

> projectsUrl: Added yesterday through [this PR](https://github.com/elastic/kibana/pull/163076/files#diff-ed856825e1d872d57b84908c2b0523bee3b567df3cb5f98188119010f4493d88). This one seems to be serverless specific, and since 8.10.0 hasn't been released yet, perhaps we can challenge the fact that it is not under the serverless object. CC @Dosant WDYT? BTW the property is not present in the README.md.

I don't mind, your call. 
But keep in mind that we will need to update project-controller. 

The reason I didn't put it under serverless initially called out here: 
https://github.com/elastic/kibana/pull/163076/files#r1283288347

I wanted to have the default value and be able to test locally, but to add serverless.* config we require to also add serverless.projectid. So I couldn't just add serverless.project_url for testing without serverless.projectid

Reply:
TBH I can see @Dosant reasoning, since it is also already being used in project controller and one could argue this is under the concern of Cloud (since it is a cloud URL) I am happy with it living outside `serverless` -- i.e., leave it as it is now.

(also already being used in [project controller](https://github.com/elastic/project-controller/pull/442))

Let's call it here and commit to placing any future values inside of `serverless`. WDYT @Dosant @gsoldevila ?

Reply:
My bad, my brain read `projectUrl` as if it was a URL of a single project.
That's why for me it was obvious that it should go under `serverless` config.